internal/command/jsonformat/diff.go

```diff
@@ -56,9 +60,51 @@ func precomputeDiffs(plan Plan, mode plans.Mode) diffs {
 	for _, change := range plan.ResourceChanges {
 		schema := plan.getSchema(change)
 		structuredChange := structured.FromJsonChange(change.Change, attribute_path.AlwaysMatcher())
+
+		before := []jsonplan.ActionInvocation{}
+		after := []jsonplan.ActionInvocation{}
+
+		for _, action := range plan.ActionInvocations {
+			if action.TriggeringResourceAddress != change.Address {
+				continue
+			}
+
+			switch action.TriggerEvent {
+			case "BeforeCreate", "BeforeUpdate", "BeforeDestroy":
+				before = append(before, action)
+			case "AfterCreate", "AfterUpdate", "AfterDestroy":
+				after = append(after, action)
```

Comment:
Can you put constants for these values somewhere in the `jsonplan` package? Or wherever the source actually is?